snapshot/snapshotter.go

```diff
@@ -296,6 +296,15 @@ type Snapshotter interface {
 	// Walk all snapshots in the snapshotter. For each snapshot in the
 	// snapshotter, the function will be called.
 	Walk(ctx context.Context, fn func(context.Context, Info) error) error
+
+	// Close releases the internal resources and makes it so that a new
+	// snapshotter instance can be created using the same state directory.
+	//
+	// The most expected usecase of this function is unit testing.
```

Comment:
I don't think this should be the primary use case of a close function. Should this not be the lifecycle endpoint of the snapshotter?